huaweicloud/helper/schemas/schema_wrapper_test.go

```diff
@@ -0,0 +1,477 @@
+	assert.Equal(t, nil, w.PrimToArray("money", false))
+
+	assert.Equal(t, nil, w.PrimToArray("gender"))
+	assert.Equal(t, []bool{}, w.PrimToArray("gender", true))
+	assert.Equal(t, nil, w.PrimToArray("gender", false))
+
+	assert.Equal(t, nil, w.PrimToArray("foo"))
+	assert.Equal(t, []any{}, w.PrimToArray("foo", true))
+	assert.Equal(t, nil, w.PrimToArray("foo", false))
+
+	assert.Equal(t, nil, w.PrimToArray("vpcs"))
+	assert.Equal(t, []any{}, w.PrimToArray("vpcs", true))
+	assert.Equal(t, nil, w.PrimToArray("vpcs", false))
+
+	// init data
+	InitTestStruct(t, d)
+
+	assert.Equal(t, []string{"jack"}, w.PrimToArray("name"))
+	assert.Equal(t, []string{"jack"}, w.PrimToArray("name", true))
+	assert.Equal(t, []string{"jack"}, w.PrimToArray("name", false))
+
+	assert.Equal(t, []int{20}, w.PrimToArray("age"))
+	assert.Equal(t, []int{20}, w.PrimToArray("age", true))
+	assert.Equal(t, []int{20}, w.PrimToArray("age", false))
+
+	assert.Equal(t, []float64{123.45}, w.PrimToArray("money"))
+	assert.Equal(t, []float64{123.45}, w.PrimToArray("money", true))
+	assert.Equal(t, []float64{123.45}, w.PrimToArray("money", false))
+
+	assert.Equal(t, []bool{true}, w.PrimToArray("gender"))
+	assert.Equal(t, []bool{true}, w.PrimToArray("gender", true))
+	assert.Equal(t, []bool{true}, w.PrimToArray("gender", false))
+
+	expectedResult := []interface{}{
+		map[string]interface{}{"beta1": "aaa", "beta2": "ccc", "beta3": "bbb"},
+	}
+	assert.Equal(t, expectedResult, w.PrimToArray("beta"))
+	assert.Equal(t, expectedResult, w.PrimToArray("beta", true))
+	assert.Equal(t, expectedResult, w.PrimToArray("beta", false))
+}
+
+func TestListToSlice(t *testing.T) {
+	// create empty struct
+	d := CreateTestStruct().TestResourceData()
+	w := NewSchemaWrapper(d)
+
+	// test nil
+	assert.Equal(t, nil, w.ListToSlice("name", false, func(_ int) any { return d.Get("name") }))
+	assert.Equal(t, nil, w.ListToSlice("beta", false, func(_ int) any { return d.Get("beta") }))
+	assert.Equal(t, nil, w.ListToSlice("vpcs", false, func(i int) any { return d.Get(fmt.Sprintf("vpcs.%d", i)) }))
+	assert.Equal(t, nil, w.ListToSlice("foo", false, func(i int) any { return d.Get(fmt.Sprintf("foo.%d", i)) }))
+
+	assert.Equal(t, []interface{}{}, w.ListToSlice("name", true, func(_ int) any { return d.Get("name") }))
+	assert.Equal(t, []interface{}{}, w.ListToSlice("beta", true, func(_ int) any { return d.Get("beta") }))
+	assert.Equal(t, []interface{}{}, w.ListToSlice("vpcs", true, func(i int) any {
+		return d.Get(fmt.Sprintf("vpcs.%d", i))
+	}))
+	assert.Equal(t, []interface{}{}, w.ListToSlice("foo", true, func(i int) any {
+		return d.Get(fmt.Sprintf("foo.%d", i))
+	}))
+
+	// init data
+	InitTestStruct(t, d)
+	assert.Equal(t, nil, w.ListToSlice("name", false, func(_ int) any { return d.Get("name") }))
+	assert.Equal(t, nil, w.ListToSlice("beta", false, func(_ int) any { return d.Get("beta") }))
+	assert.Equal(t, nil, w.ListToSlice("beta", false, func(_ int) any { return d.Get("beta") }))
```

Comment:
This line of code is duplicated with line 252.

Reply:
fix